graphql_compiler/interpreter/expression_ops/immediate_expression_handlers.py

```diff
@@ -25,6 +25,10 @@ def evaluate_local_field(
     expression: LocalField,
     data_contexts: Iterable[DataContext],
 ) -> Iterator[Tuple[DataContext, Any]]:
+    if current_location is None:
+        raise AssertionError(f"Received None for current_location, which indicates we are "
+                             f"in the global scope, but there can't be any local fields there.")
```

Comment:
```suggestion
        raise AssertionError(
            f"Unexpectedly attempting to evaluate a LocalField while in the global scope "
            f"(current_location is None). This is a bug: {expression}"
        )
```

Reply:
Good fix, thanks. With the way I wrote it, a user would have no idea if this is a bug or an obscure user error.